Source/Document Structure/SvgLink.cs

```diff
@@ -0,0 +1,111 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Net;
+using System.Text;
+
+namespace Svg
+{
+    [NonSvgElementAttribute("link")]
+    public class SvgLink : NonSvgElement
+    {
+        public SvgLink()
+          : base("link")
+        {
+        }
+        public enum RelativeValue
+        {
+            Unknown,
+            Alternate,
+            Author,
+            Help,
+            Icon,
+            License,
+            Next,
+            Pingback,
+            Preconnect,
+            Prefetch,
+            Preload,
+            Prerender,
+            Prev,
+            Search,
+            Stylesheet
+        }
+
+        public override SvgElement DeepCopy()
+        {
+            return DeepCopy<SvgLink>();
+        }
+
+        [SvgAttribute("href")]
+        public string Href
+        {
+            get { return GetAttribute<string>("href", false, string.Empty); }
+            set { Attributes["href"] = value; }
+        }
+
+        [SvgAttribute("rel")]
+        public RelativeValue Rel
+        {
+            get { return GetAttribute<RelativeValue>("rel", false, RelativeValue.Unknown); }
+            set { Attributes["rel"] = value; }
+        }
+
+        public string GetLinkContentAsText(RelativeValue rel)
+        {
+            try
+            {
+                using (var stream = GetLinkContentAsStream(rel))
+                {
+                    using (StreamReader sr = new StreamReader(stream))
+                    {
+                        return sr.ReadToEnd();
+                    }
+                }
+            }
+            catch
+            {
+                return string.Empty;
+            }
+        }
+
+        public Stream GetLinkContentAsStream(RelativeValue rel = RelativeValue.Unknown)
+        {
+            if (Rel != rel && rel != RelativeValue.Unknown)
+                return null;
+            // Uri MaxLength is 65519 (https://msdn.microsoft.com/en-us/library/z6c2z492.aspx)
+            // if using data URI scheme, very long URI may happen.
+            var safeUriString = Href.Length > 65519 ? Href.Substring(0, 65519) : Href;
+
+            try
+            {
+                var uri = new Uri(safeUriString, UriKind.RelativeOrAbsolute);
+
+                if (!uri.IsAbsoluteUri)
+                    uri = new Uri(OwnerDocument.BaseUri, uri);
+
+                // should work with http: and file: protocol urls
+                var httpRequest = WebRequest.Create(uri);
```

Comment:
This should check that `SvgDocument.ResolveExternalResources` is true before creating a web request to avoid XSS vulnerabilities. See #873.

Reply:
@kimsey0 Can you fix it by editing the PR source?

Reply:
It's been too long since I worked on this project, so I can't really remember the details, nor how tests etc. are structured. I can see that #873 split `SvgDocument.ResolveExternalResources` into three properties: `ResolveExternalXmlEntites`, `ResolveExternalImages`, and `ResolveExternalElements`, probably to give more fine-grained control over which external resources are allowed to be resolved. You may want to introduce a new `ExternalType ResolveExternalStylesheets` property, somehow pass it in or make it available to this `GetLinkContentAsStream` method, and then trace a warning and return null or an empty string if `!ResolveExternalStylesheets.AllowsResolving(uri)`.

Reply:
@kimsey0 Thank for the update. Just do not know the best way to handle these staled PR.
- Take over (clone PR reposition and fix issues), the original owner looses his/her contribution
- Close the PR and see how to reimplement the issue, a waste but may be worth it

Reply:
- adapt/finish the original PR, so it will just have 2 authors

Reply:
> * adapt/finish the original PR, so it will just have 2 authors

Never done, can you outline the procedure and give me some hints?

Reply:
If you are asking about the git workflow, something like:
> git remote add FabioFerrettiMoodys https://github.com/FabioFerrettiMoodys/SVG
> git checkout FabioFerrettiMoodys/master (probably under some other name)
> git rebase origin/main -> resolve conflicts as usual
> git push --force-with-lease
(or the equivalents if you use a git GUI client(
- make your changes, commit, push...

Is this what you are asking?

Reply:
> Is this what you are asking?

I think so, thanks - will look into it this weekend.

Reply:
Depending on the policies of this project (the contribution guide doesn't say), you can also just avoid squashing the commits, leaving the original author for each one and making it clear who's done what.

Reply:
> the contribution guide doesn't say

Something to be added...

The current (undocumented) convention is that commits are usually squashed and only rebased if the individual commits are independent changes to be preserved, and have sensible commit comments (which is not the case here). Though it is possible to rebase the PR respectively before the merge, if a rebase is wanted.